core/src/KokkosExp_MDRangePolicy.hpp

```diff
@@ -327,6 +327,12 @@ struct MDRangePolicy : public Kokkos::Impl::PolicyTraits<Properties...> {
     }
     for (int i = rank_start; i != rank_end; i += increment) {
       const index_type length = m_upper[i] - m_lower[i];
+
+      if (m_upper[i] < m_lower[i])
+        Kokkos::abort(
+            "Kokkos::MDRangePolicy bounds error: One of the lower bounds is "
+            "greater than the upper bound of its rank.");
```

Comment:
i would also print the index so that we can tell users more explicitly which bound was triggering this

Reply:
Modified the wording to print the input bounds and the rank. See 0e9ee83.